pkg/bosh/converter/container_factory.go

```diff
@@ -24,7 +24,13 @@ const (
 )
 
 // ContainerFactory builds Kubernetes containers from BOSH jobs.
-type ContainerFactory struct {
+type ContainerFactory interface {
+	JobsToInitContainers(jobs []bdm.Job, defaultVolumeMounts []corev1.VolumeMount, bpmDisks BPMResourceDisks) ([]corev1.Container, error)
+	JobsToContainers(jobs []bdm.Job, defaultVolumeMounts []corev1.VolumeMount, bpmDisks BPMResourceDisks) ([]corev1.Container, error)
+}
+
+// ConcreteContainerFactory is a concrete implementation of ContainerFactor.
+type ConcreteContainerFactory struct {
```

Comment:
`ContainerFactoryImpl` seems to be a common naming in Golang (Java?). While the `Impl` suffix doesn't add much, I don't think 'Concrete', 'Base' or 'Default' are any better ;)

This naming problem goes away if the the interface is declared in the consumer's package. Otherwise, can we find a more descriptive name? "K8SContainerFactory", "NotFakeContainerFactory", ugh no, apparently I can't :)
We only add it for mocking anyhow and I don't see how there could be another implementation of that factory.

Reply:
Thx. I decided to move interfaces to consumer's package.